tracing-appender/src/rolling.rs

```diff
@@ -100,20 +99,20 @@ pub struct RollingFileAppender {
 #[derive(Debug)]
 pub struct RollingWriter<'a> {
     inner: RwLockReadGuard<'a, File>,
-    current_size: &'a AtomicU64,
+    current_size: &'a AtomicUsize,
 }
 
 #[derive(Debug)]
 struct Inner {
     log_directory: PathBuf,
     log_filename_prefix: Option<String>,
     log_filename_suffix: Option<String>,
-    log_filename_index: Option<AtomicU64>,
+    log_filename_index: AtomicUsize,
     date_format: Vec<format_description::FormatItem<'static>>,
     rotation: Rotation,
     next_date: AtomicUsize,
-    current_size: AtomicU64,
     max_files: Option<usize>,
+    current_size: AtomicUsize,
```

Comment:
I think we should use `AtomicU64`, not `AtomicUsize`.
`usize` is used for memory indexing (that’s why it’s pointer-sized), but IIRC we never these variables it for this purpose here.
For instance, in the standard library, file length is returned using a u64: https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.len

Reply:
I did that because the current size just uses the `Write::write()` output, which returns a `usize`. I mean, conceptually a `u64` would make more sense as you said, but in practice, it looks unnecessary to use a `u64` if we are storing a `usize` to compare it to another `usize` right?

Reply:
Ultimately I don’t think it will cause any problem because the point of using a rolling appender is to avoid big files, but the maximum size of a file is not bounded by `usize` and we are not just comparing but also accumulating the current size of the file which may grow bigger than the maximum value of `usize` on some platforms. You may be right, but at the same time the less we take this kind of bet, the less we expose ourselves to weird surprising edge cases. Ultimately the PR will be reviewed again by a tokio member, so I’m fine with leaving that as-is and waiting for their input too.

However for `log_filename_index` I don’t think we compare it with an `usize`  anywhere so there is no ergonomic benefit for this one. Again, it’s likely fine to use an `usize`, but the appropriate type would be a `u64`, so why not use it?

Reply:
That makes total sense, thanks for all the context! Let me change it back to `u64` since it's the correct type to use as you said and we can leave the rest for the tokio reviewers.